tests/phpunit/test-fonts.php

```diff
@@ -0,0 +1,216 @@
+		$font_dir = wp_get_font_dir();
+		
+		$expected = [
+			'path' => WP_CONTENT_DIR . '/uploads/fonts',
+			'url' => WP_CONTENT_URL . '/uploads/fonts',
+			'subdir' => '',
+			'basedir' => WP_CONTENT_DIR . '/uploads/fonts',
+			'baseurl' => WP_CONTENT_URL . '/uploads/fonts',
+			'error' => false,
+		];
+
+		$this->assertEquals( $expected, $font_dir );
+	}
+
+	/**
+	 * Test that using the font_dir filter at priority 10 overrides our modifications.
+	 */
+	public function test_pantheon_fonts_dir_filter() {
+		// Check current WP version to see if we can run the test.
+		$version = _pantheon_get_current_wordpress_version();
+		if ( version_compare( $version, '6.4.x', '<=' ) ) {
+			// Skip the test if the current WP version is less than 6.5.
+			$this->markTestSkipped( 'WP 6.5+ or Gutenberg 17.6+ must be available to test the font library modifications.' );
+		}
+
+		$this->maybe_get_font_library();
+		if ( ! function_exists( 'wp_get_font_dir' ) ) {
+			// If the function still doesn't exist after trying to get the font library from gutenberg, mark the test skipped.
+			$this->markTestSkipped( 'The wp_get_font_dir function is not available. We\'re probably not using WP 6.5+' );
+		}
+
+		$custom_directory = [
+			'path' => WP_CONTENT_DIR . '/custom-fonts',
+			'url' => WP_CONTENT_URL . '/custom-fonts',
+			'basedir' => WP_CONTENT_DIR . '/custom-fonts',
+			'baseurl' => WP_CONTENT_URL . '/custom-fonts',
+		];
+
+		add_filter( 'font_dir', function ( $defaults ) use ( $custom_directory ) {
+			$defaults['path'] = $custom_directory['path'];
+			$defaults['url'] = $custom_directory['url'];
+			$defaults['basedir'] = $custom_directory['basedir'];
+			$defaults['baseurl'] = $custom_directory['baseurl'];
+			return $defaults;
+		} );
+
+		Fonts\bootstrap();
+		$font_dir = wp_get_font_dir();
+
+		$expected = [
+			'path' => WP_CONTENT_DIR . '/custom-fonts',
+			'url' => WP_CONTENT_URL . '/custom-fonts',
+			'subdir' => '',
+			'basedir' => WP_CONTENT_DIR . '/custom-fonts',
+			'baseurl' => WP_CONTENT_URL . '/custom-fonts',
+			'error' => false,
+		];
+
+		$this->assertEquals( $expected, $font_dir );
+	}
+
+	/**
+	 * Test that the font directory modifications can be disabled.
+	 */
+	public function test_disable_pantheon_font_dir_mods() {
+		// Check current WP version to see if we can run the test.
+		$version = _pantheon_get_current_wordpress_version();
+		if ( version_compare( $version, '6.4.x', '<=' ) ) {
+			// Skip the test if the current WP version is less than 6.5.
+			$this->markTestSkipped( 'WP 6.5+ or Gutenberg 17.6+ must be available to test the font library modifications.' );
+		}
+
+		$this->maybe_get_font_library();
+		if ( ! function_exists( 'wp_get_font_dir' ) ) {
+			// If the function still doesn't exist after trying to get the font library from gutenberg, mark the test skipped.
+			$this->markTestSkipped( 'The wp_get_font_dir function is not available. We\'re probably not using WP 6.5+' );
+		}
```

Comment:
Lot of repetition in these tests. maybe not right to use dataloader, but seems like it could be refined.